src/System.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/ParametersTest.cs

```diff
@@ -98,6 +98,20 @@ public static void CodeCoverageSqlClient()
         }
 
         [CheckConnStrSetupFact]
+        public static void Test_SqlParameter_Constructor()
+        {
+            using (var conn = new SqlConnection(s_connString))
+            {
+                conn.Open();
+                var cmd = new SqlCommand("select @input as 'test'", conn);
+                var sqlParameter = new SqlParameter("@input", SqlDbType.Int, 1, ParameterDirection.Input, false, 1, 0, "test", DataRowVersion.Current, MyEnum.A);
+                cmd.Parameters.Add(sqlParameter);
```

Comment:
Can the test validate all the parameters -- eg., sourceColumn? 

Sorry to be picky!

Reply:
Not a problem, I've updated the test to use an adapter and datatable to test sourceColumn and all other parameters. Not sure if there's a better way to do this, but it seemed like a thorough (and common) use case.